config/routes.rb

```diff
@@ -19,8 +19,9 @@
   get  "/login/oauth/authorize", to: "oauth#authorize"
   post  "/login/oauth/access_token", to: "oauth#access_token"
 
-  match "/auth/:provider/callback", to: "sessions#create",  via: %i[get post]
-  match "/auth/failure",            to: "sessions#failure", via: %i[get post]
+  match "/auth/lti/callback",       to: "sessions#lti_launch",  via: %i[get post]
```

Comment:
Hello from Appsec!

I was hoping to get some clarification as to why we're accepting both `GET` and `POST` requests to these three routes. 

* Supporting `GET` requests will bypass the `verify_authenticity_token` checks, and leave the application potentially vulnerable to [CSRF](https://github.com/github/security-docs/blob/master/standards/Secure%20Coding%20Principles.md#csrf-tokens), which likely is not a concern for these integrations.
* Supporting `POST` requests is preferred by default because it enables us to pass sensitive data in the body parameters as to not leak the values in the query string. This would also execute the `verify_authenticity_token` to protect from CSRF. My concern here is that the LTI integration shouldn't have knowledge of the user's `authenticity token` to provide and as such, likely would raise an exception.

I guess my question is: Do we *need* to support both `GET` and `POST` requests to all three of these routes? Or is this just a quick catch-all?